fix: resolve EPLUGINSCONF error by using CommonJS module.exports pattern

LittleCoinCoin opened the pull request:
### Description

Fixes the EPLUGINSCONF error that prevents semantic-release from loading the plugin in production environments.

**Closes #3**

### Changes Made

1. **Core Fix**: Convert from ES6 named exports to CommonJS `module.exports` pattern
   - Changes `export async function` to `module.exports = { verifyConditions, prepare }`
   - Ensures semantic-release receives a plain object instead of ES module interop wrapper

2. **Build Configuration**: Update Vite config for semantic-release compatibility
   - Use `formats: ['cjs']` only (remove ES module output)
   - Add explicit `output: { format: 'cjs', exports: 'auto' }`

3. **Cross-platform Compatibility**: Replace `rm -rf` with `rimraf` for Windows support
   - Fixes build failures on Windows environments
   - Ensures CI/CD pipeline compatibility across platforms

### Technical Details

**Before (broken):**
```javascript
// Generated output
exports.prepare=f;exports.verifyConditions=y;
// Creates: Object [Module] { prepare: [AsyncFunction], verifyConditions: [AsyncFunction] }
```

**After (working):**
```javascript
// Generated output  
module.exports={verifyConditions:y,prepare:f};
// Creates: { verifyConditions: [AsyncFunction], prepare: [AsyncFunction] }
```

### Testing

- ✅ Plugin loads successfully with semantic-release v22.0.12
- ✅ No EPLUGINSCONF errors in dry-run mode
- ✅ Cross-platform build compatibility verified
- ✅ Maintains all existing functionality

### Verification

```bash
# Test plugin loading
npx semantic-release --dry-run --no-ci

# Expected output:
# [semantic-release] » √  Loaded plugin "verifyConditions" from "@artessan-devs/sr-uv-plugin"
# [semantic-release] » √  Loaded plugin "prepare" from "@artessan-devs/sr-uv-plugin"
```

### Breaking Changes

None. This is a pure bug fix that maintains the same API and functionality.

Commit:
fix: resolve EPLUGINSCONF error by using CommonJS module.exports

- Convert from ES6 named exports to CommonJS module.exports pattern
- Update Vite config to generate CommonJS-only output with explicit export control
- Semantic-release expects plain objects, not ES module interop wrappers
- Build output changes from 'exports.prepare=f' to 'module.exports={prepare:f}'

Resolves semantic-release plugin loading issue that prevented deployment.

Commit:
fix: replace rm command with cross-platform rimraf for Windows compatibility

- Replace 'rm -rf dist' with 'rimraf dist' in clean script
- Add rimraf as devDependency for cross-platform file deletion
- Ensures build process works on Windows, Linux, and macOS environments
- Resolves CI/CD pipeline compatibility issues

Artessan commented on package.json:
```diff
@@ -33,6 +33,7 @@
   "license": "MIT",
   "devDependencies": {
     "@types/node": "^22.15.3",
+    "rimraf": "^6.0.1",
```
rimraf is a heavy lib just for a command, we can write or own 
`"clean": "node -e \"import('fs/promises').then(fs => fs.rm('dist', { recursive: true, force: true }))\""`

or replace with [shx](https://www.npmjs.com/package/shx) that is a small dependency and principal uses are cases like this.

LittleCoinCoin replied:
Thanks for the review!
I agree with limiting the dependencies; I will check with the code exec you are suggesting.

Artessan reviewed (changes requested):
Thank you! The only blocker is the use of rimraf. You can update the PR with one of the suggested alternatives or undo that change, and we can merge it. If not, please just close the PR and I’ll create a new one with the changes.

And again, thanks!

Commit:
feat: replace rimraf with Node.js built-in fs.rm()

- Remove rimraf dependency from devDependencies
- Update clean script to use Node.js fs.rm() with recursive and force options
- Maintains cross-platform compatibility (Node 14.14.0+)
- Reduces dependency footprint while preserving functionality

LittleCoinCoin commented:
Hello!

Thank you for the feedback, I've made the change using your suggested code snippet. But I wanted to bring up an issue I discovered during testing.

The tests are now failing when trying to import from the source (`import { prepare, verifyConditions } from '../src/semantic-release-pyproject'`), but they pass when importing from the built distribution (`const { prepare, verifyConditions } = require('../dist/index.js')`).

I suspect this is happening because we chose to follow CommonJS exports to work with semantic-release, but Vite apparently only supports TypeScript ES modules during development...? The build process seems to add ES module interop code that semantic-release can't handle, which is why we had to switch to testing against the distribution.

I'd recommend you don't merge this PR until you can test it locally and see if you experience the same issue. The core functionality works fine - it seems to be a testing/build configuration conflict.

A few questions for you:
1. Do you see the same test failures when running the original source-based tests?
2. Would you prefer to keep the original testing approach and find a different solution?
3. Are you open to a hybrid testing strategy (source for development, distribution for production validation)?

Happy to discuss further or leave it up to you.

Artessan commented:
Hey,

Thanks for the changes. Just ended of testing locally and the same problems present to me, so I think as a possible solution we can just use the ES exports and add an index file with  only the module.export in that way it will be exported correctly as a common module and we will still be able to use the ES in testing

LittleCoinCoin committed:
refactor: implement ES6/CommonJS hybrid solution with index wrapper

- Revert source file to ES6 exports for test compatibility
- Add src/index.ts as CommonJS adapter layer
- Update vite config to use index.ts as entry point
- Tests now pass importing from source with ES6 syntax
- Build output maintains CommonJS format for semantic-release

This implements the maintainer's suggested approach: ES exports in
source files with module.exports wrapper in index file.

LittleCoinCoin commented:
Hello @Artessan !

I've implemented the ES6/CommonJS hybrid solution exactly as you proposed (I think...)

**Changes:**
- Reverted `src/semantic-release-pyproject.ts` to ES6 exports
- Created `src/index.ts` as a CommonJS wrapper with `module.exports`
- Updated vite config to use the index file as entry point

As a result tests are passing on my side. Please, tell me how it goes for you :)

LittleCoinCoin commented:
Hello @Artessan !
Any feedback on this? I was doing some deps management on some of my project and realised I am still pointing to my fork rather than your repo ;)
